snapshots/overlay/overlay.go

```diff
@@ -518,3 +624,53 @@ func supportsIndex() bool {
 	}
 	return false
 }
+
+// supportsMetacopy checks whether the "metacopy" option is supported by the kernel.
+func supportsMetacopy() bool {
+	if _, err := os.Stat("/sys/module/overlay/parameters/metacopy"); err != nil {
+		logrus.Infof("Metacopy unsupported.")
+		return false
+	}
+	return true
+}
+
+// isMetacopyOn checks whether the "metacopy" option is on/off by kernel.
+// the caller should call supportsMetacopy first. isMetacopyOn returns true
+// if "metacopy=on", if "metacopy=off" or metacopy not supported it returns false.
+func isMetacopyOn() bool {
+	content, err := os.ReadFile("/sys/module/overlay/parameters/metacopy")
+	if err != nil {
+		return false
+	}
+
+	if bytes.Equal(content, []byte("Y\n")) {
+		logrus.Infof("Metacopy enabled by the kernel.")
+		return true
+	}
+
+	logrus.Infof("Metacopy disabled by the kernel.")
+	return false
+}
+
+// cacheSupportsMetacopy checks whether the "metacopy" option has been enabled/disabled earlier
+func cacheSupportsMetacopy(root string) metacopyCacheFlag {
+	// metacopy disabled
+	if _, err := os.Stat(filepath.Join(root, "metacopy-false")); err == nil {
+		return metacopyCacheFalse
+	}
+	// metacopy enabled
+	if _, err := os.Stat(filepath.Join(root, "metacopy-true")); err == nil {
```

Comment:
I don't think having different files is a good idea. This allows, for example, to both files to be present which is not a nice scenario we want to be at.

It seems better to just have one file and, like the same way we do for `/sys/module/overlay/parameters/metacopy`, make the file content be Y/N and that is all.

Reply:
Yeah, should look better, thanks!